patches/server/0703-Add-Raw-Byte-Entity-Serialization.patch

```diff
@@ -58,7 +38,8 @@ index 0a3d447f87698dc786d6cab6ded27eb1b5780204..d3b37adfe6d66e82db18d94f143af3ab
 +        Preconditions.checkArgument(entity instanceof org.bukkit.craftbukkit.entity.CraftEntity, "only CraftEntities can be serialized");
 +
 +        CompoundTag compound = new CompoundTag();
-+        ((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandle().serializeEntity(compound);
++        compound.putString("id", net.minecraft.world.entity.EntityType.getKey(((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandle().getType()).toString());
++        ((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandle().saveWithoutId(compound);
```

Comment:
This would serialise passengers of the entity serialised which breaks the current return contract.
There is a reason the current logic sets the passengers to an empty list before calling save()

Reply:
Passengers are not preserved upon spawning the deserialized entity, but true, this data should probably omit those values immediately upon serialization.
Brought back the workaround 👍